src/EditorFeatures/Core.Wpf/WpfClassificationExtensions.cs

```diff
@@ -18,12 +18,17 @@ internal static partial class WpfClassificationExtensions
     {
         public static Run ToRun(this ClassifiedText part, IClassificationFormatMap formatMap, ClassificationTypeMap typeMap)
         {
-            var text = part.Text;
+            var run = new Run(part.Text);
 
-            var run = new Run(text);
+            var classificationType = typeMap.GetClassificationType(part.ClassificationType);
```

Comment:
:bulb: This method is mixing the expected case with code for an impossible¹ error condition. It would improve readability to extract a helper function to get a classification type, and include the fallback behavior there instead.

¹ We have tests that are supposed to eliminate this case, but it sounds like something still failed.